mongo/instance_store.go

```diff
@@ -121,30 +112,31 @@ func (m *Mongo) UpdateInstance(ctx context.Context, currentInstance, updatedInst
 	}
 	updatedInstance.ETag = newETag
 
-	// create selector query
-	sel := selector(currentInstance.InstanceID, updatedInstance.UniqueTimestamp, eTagSelector)
-
 	// create update query from updatedInstance and newly generated eTag
 	updates := createInstanceUpdateQuery(ctx, currentInstance.InstanceID, updatedInstance)
 	update := bson.M{"$set": updates}
-	updateWithTimestamps, err := dpmongo.WithUpdates(update)
+	updateWithTimestamps, err := mongodriver.WithUpdates(update)
 	if err != nil {
 		return "", err
 	}
 
+	// create selector query
+	sel := selector(currentInstance.InstanceID, updatedInstance.UniqueTimestamp, eTagSelector)
+
 	// execute the update against MongoDB to atomically check and update the instance
-	if err = s.DB(m.Database).C(instanceCollection).Update(sel, updateWithTimestamps); err != nil {
-		if err != mgo.ErrNotFound {
-			return "", err
+	if _, err = m.Connection.C(instanceCollection).Must().Update(ctx, sel, updateWithTimestamps); err != nil {
```

Comment:
Just a question, does `Must` panic if there is some error? I followed the code and I think it only returns errors, which is fine but wanted to double-check (the library code shouldn't panic)

Reply:
Yes it will not panic. The Must wrapper is to ensure that it will also return a `no document found` error if update selector does not find a document to update